### PR TITLE
[Catalog] Switch NSMutableArray to Swift Array.

### PR DESCRIPTION
**Pull request by kolodny**
This PR is mostly a cleanup to use a Swift array instead of an NSMutableArray

### CODE REVIEW

File: catalog/MDCCatalog/MDCCatalogComponentsController.swift [hunk 1]
```diff
@@ -29,7 +29,6 @@ class MDCCatalogComponentsController: UICollectionViewController, MDCInkTouchCon
   let inset = CGFloat(16)
   let node: CBCNode
   var headerViewController: MDCFlexibleHeaderViewController
-  let imageNames = NSMutableArray()
```
Comment (kolodny) [inline catalog/MDCCatalog/MDCCatalogComponentsController.swift:32]: This didn't appear to be used anywhere

File: catalog/MDCCatalog/MDCNodeListViewController.swift [hunk 2]
```diff
@@ -62,22 +62,19 @@ class MDCNodeListViewController: CBCNodeListViewController {
...
-          componentDescription = childNode.exampleDescription()
-        } else {
-          orderedNodes.add(childNode)
-        }
-      }
+    if let primaryDemoNodeIndex = childrenNodes.index(where: { $0.isPrimaryDemo() }),
+        primaryDemoNodeIndex != 0 {
+      let primaryDemoNode = childrenNodes[primaryDemoNodeIndex]
+      childrenNodes.remove(at: primaryDemoNodeIndex)
+      childrenNodes.insert(primaryDemoNode, at: 0)
```
Comment (willlarche) [inline catalog/MDCCatalog/MDCNodeListViewController.swift:72]: I saw a one-line way to move elements:
`insert(remove(at: primaryDemoNodeIndex), at: 0)`

**Review by willlarche — APPROVED** [review]

File: catalog/MDCCatalog/MDCNodeListViewController.swift (hunk 2 above)
Comment (kolodny) [reply catalog/MDCCatalog/MDCNodeListViewController.swift:72]: I find that snippet somewhat too clever, what do you think about the following?

```swift
let primaryDemoNode = childrenNodes.remove(at: primaryDemoNodeIndex)
childrenNodes.insert(primaryDemoNode, at: 0)
```

Comment (willlarche) [reply catalog/MDCCatalog/MDCNodeListViewController.swift:72]: I'll take it. I just wish the language had 'move'

**Review by romoore — APPROVED** [review]
LGTM

**Pull request merged**